tests/js/WithRegistrySetup.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * WithRegistrySetup component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+const { useRegistry } = Data;
+
+const WithRegistrySetup = ( { func, children } ) => {
+	const registry = useRegistry();
+	const [ ready, setReady ] = useState( false );
+
+	useEffect( () => {
```

Comment:
Looks like there's a linting error here, probably good to clear up 🙂

Reply:
@tofumatt Originally there was a linting error which I fixed. Do you still see the error? All the tests are passing for me as well as the Github checks.

Reply:
This should probably use `useMount` instead of `useEffect` like we do in other places now when an effect should only run once. I think the linting error here is from the missing dependency array though below.